lib/android/app/src/main/java/com/reactnativenavigation/anim/StackAnimator.java

```diff
@@ -109,40 +108,35 @@ public void onAnimationRepeat(Animator animation) {
 	private float getWindowHeight(Context context) {
 		DisplayMetrics metrics = new DisplayMetrics();
 		WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
-		windowManager.getDefaultDisplay().getMetrics(metrics);
+		if (windowManager != null) {
+			windowManager.getDefaultDisplay().getMetrics(metrics);
+		}
 		return metrics.heightPixels;
 	}
 
-	public void animateShowTopBar(final TopBar topBar, final View container) {
-		ValueAnimator containerHeightAnim = ValueAnimator.ofInt(container.getMeasuredHeight(), container.getMeasuredHeight() - topBar.getMeasuredHeight());
-		containerHeightAnim.setInterpolator(DECELERATE_INTERPOLATOR);
-		containerHeightAnim.setDuration(DURATION_TOPBAR);
-		containerHeightAnim.addUpdateListener(valueAnimator -> {
-            int val = (Integer) valueAnimator.getAnimatedValue();
-			ViewGroup.LayoutParams layoutParams = container.getLayoutParams();
-            layoutParams.height = val;
-            container.setLayoutParams(layoutParams);
-        });
-		ObjectAnimator containerTransitionAnim = ObjectAnimator.ofFloat(container, View.TRANSLATION_Y, -1 * topBar.getMeasuredHeight(), 0);
-		containerTransitionAnim.setInterpolator(DECELERATE_INTERPOLATOR);
-		containerTransitionAnim.setDuration(DURATION_TOPBAR);
+    public void animateShowTopBar(final TopBar topBar, final View container) {
+        animateShowTopBar(topBar, container, -1 * topBar.getMeasuredHeight());
+    }
 
-		ObjectAnimator topbarAnim = ObjectAnimator.ofFloat(topBar, View.TRANSLATION_Y, -1 * topBar.getHeight(), 0);
-		topbarAnim.setInterpolator(DECELERATE_INTERPOLATOR);
-		topbarAnim.setDuration(DURATION_TOPBAR);
+    public void animateShowTopBar(final TopBar topBar, final View container, float startTranslation) {
+        ObjectAnimator topbarAnim = ObjectAnimator.ofFloat(topBar, View.TRANSLATION_Y, startTranslation, 0);
+        topbarAnim.setInterpolator(DECELERATE_INTERPOLATOR);
+        topbarAnim.setDuration(DURATION_TOPBAR);
 
-		AnimatorSet set = new AnimatorSet();
-		set.addListener(new Animator.AnimatorListener() {
+		topbarAnim.addListener(new Animator.AnimatorListener() {
 			@Override
 			public void onAnimationStart(Animator animation) {
 				topBar.setVisibility(View.VISIBLE);
 			}
 
 			@Override
 			public void onAnimationEnd(Animator animation) {
-                ViewGroup.LayoutParams layoutParams = container.getLayoutParams();
-				layoutParams.height = ViewGroup.LayoutParams.MATCH_PARENT;
-				container.setLayoutParams(layoutParams);
+				//TODO: needs refactoring
```

Comment:
Either refactor now, or remove the comment, open an issue detailing what needs to be refactored and assign your self.